### PR TITLE
samples/guestbook: add the ability to run Guestbook on Azure

### LINKED ISSUE #1122: samples/guestbook: add azure to the guestbook sample

### PR DESCRIPTION
**Pull request by vangent**
Fixes #1122.

Note that this is only a halfway step:

* We're running the binary and MySQL locally. Until we support some kind of SQL offering for Azure, we can't easily run on an VM at Azure. #1305 will fix this.

* For `runtimevar`, we're using a blob in Azure Storage, so only kind of cheating. #1314 will fix this if there's an Azure equivalent of of ParamStore/RuntimeConfigurator.

I've updated both of those bugs with a note to update Guestbook once they are fixed.

At least one reviewer (if not both) should actually follow the instructions and make sure they can successfully get it running.

### CODE REVIEW

File: samples/guestbook/README.md [hunk 1]
```diff
@@ -185,6 +188,77 @@ directory using the same variables you entered during `terraform apply`.
...
+storage_container = [redacted]
+```
+
+### Running
+
+You will need to run a local MySQL database server, similar to what we did for
+running locally earlier.
+
+```shell
+go get ./localdb/... # Get package dependencies.
```
Comment (clausti) [inline samples/guestbook/README.md]: Minor note: having previously been directed to `cd` into the azure directory, one is in the wrong directory to run `go get ./localdb/...`. Might be worth adding an instruction to go up one level?

File: samples/guestbook/README.md [hunk 2]
```diff
@@ -185,6 +188,77 @@ directory using the same variables you entered during `terraform apply`.
...
+go run localdb/main.go
+```
+
+In another terminal, add your Azure credentials to the environment and run the
+`guestbook` application:
+
+```shell
+# Enter the storage_account from the Terraform output earlier.
+AZURE_STORAGE_ACCOUNT=<your storage_account>
```
Comment (clausti) [inline samples/guestbook/README.md]: add `export` to the beginning of these lines?

**Comment by vangent** [comment]
> Got it up and running! And followed the instructions successfully all the way through `terraform destroy`.

Great, thanks!

> Not changed in this PR so I can't comment on it directly, but it might be worth adding that Docker "Desktop" is the correct tool to install.

Done.

File: samples/guestbook/README.md (hunk 1 above)
Comment (vangent) [reply samples/guestbook/README.md]: Done.

File: samples/guestbook/README.md (hunk 2 above)
Comment (vangent) [reply samples/guestbook/README.md]: Done.

**Comment by vangent** [comment]
@zombiezen PTAL

**Pull request merged**